Lib/asyncio/tasks.py

```diff
@@ -252,6 +253,9 @@ def _wakeup(self, future):
 
 
 _PyTask = Task
+_py_register_task = _register_task
+_py_enter_task = _enter_task
+_py_leave_task = _leave_task
```

Comment:
You should do this logic in `base_tasks.py`.

Reply:
Not so easy: `asyncio._asyncio` imports `asyncio.base_tasks`. C Accelerators becomes available only in `asyncio.tasks`, pushing them into `base_tasks` leads to circular imports.

That's why I've decided to have both `_py_register_task` and `_c_register_task` in the same `asyncio.tasks` module.

Reply:
Then define all these functions (and `_all_tasks` and `_current_tasks` mappings) in `_asynciomodule.c` and try to import them in `tasks.py`.  If that fails, define them in Python.  Let's not touch `base_tasks.py` at all in this PR.

Reply:
If we define `_all_tasks` in `_asynciomodule.c` -- C Accelerator becomes mandatory.
Let's define `_all_tasks` and `_current_tasks` in `base_tasks.py` but move `_register_task` and family into `tasks.py`.

Reply:
> C Accelerator becomes mandatory.

No, why?

`_asynciomodule.c`:

```
// defines _all_tasks, _current_tasks
```

`asyncio/tasks.py`:

```python
try:
    from _asyncio import _all_tasks, _current_tasks
except ImportError:
    _all_tasks = ...
    _current_tasks = ...
```

Reply:
Also keep in mind that `_all_tasks` isn't public API, it's only needed to implement `tasks.py/Task` class, so you only defined it when `_asynciomodule.c` isn't compiled.

Reply:
Aha, it works.
Another problem remains: `asyncio.current_loop()` need `_current_tasks`. On other hand the function should be available for import in `_asynciomodule.c` -- `CTask.current_loop()` calls it (I don't want duplicate the logic). Thus `asyncio.current_loop()` should be in `base_tasks.py`.

Reply:
Or we can implement `get_running_loop()` and `asyncio.current_loop()` in `_asynciomodule.c`.  Then we don't need to touch `base_tasks.py`.  I actually wanted to make a separate PR for `get_running_loop()` in C, as it's performance is really critical.  Let's do it?

Reply:
> I actually wanted to make a separate PR

I can make that PR myself now.

Reply:
working on it.

Reply:
Please do.
I'm going to sleep soon anyway.

Reply:
> Please do.

Almost there.

> Thus asyncio.current_loop() should be in base_tasks.py.

Yeah... I get what you're saying now.  OK, let it be in `base_tasks.py`.  I'm not blocking you from working on this PR, right?

Reply:
Yes, you are not blocking.